test.py

```diff
@@ -802,6 +808,16 @@ def test_csv_import(self):
         # Teardown step.
         os.remove("beautiful_table.csv")
 
+    def test_df__export(self):
+        df = self.table.to_df()
+        self.assertEqual(self.table.rows.header, df.index)
+        self.assertEqual(self.table.columns.header, list(df.columns))
+
+    def test_df_import(self):
+        table = BeautifulTable()
+        table = table.from_df(self.df)
+        self.assertEqual(self.table.rows.header, self.df.index)
```

Comment:
we should also compare the cell values, not just the headers. also maybe add other unit tests to check the behavior in case headers are missing.

Reply:
Done

Reply:
Let's not use the _data attribute in the unit tests. private variable so only meant to be used from internal modules.
[list(row) for row in list(self.table._data)] => [list(row) for row in table.rows]

Reply:
Yes, I have changed this in beautifultable.py also. do you know I used `[list(row) for row in table.rows]` while adding asdict API, don't know why I used  _data there.
No probs, I am thinking of picking the row-span, column-span, and transpose features there I will get some good exposure to the design.

Reply:
spans might be tricky. needs some design before any implementations. I did start with something but found some gaps in the approach hence had to abandon the idea. but yeah, would love to see if you come up with an approach on how to implement these on their respective threads.

Reply:
Sure thing